qiita_db/test/test_data.py

```diff
@@ -478,6 +478,38 @@ def test_create_data_type_only(self):
         # preprocessed_data_id, filepath_id
         self.assertEqual(obs, [[3, obs_id - 1], [3, obs_id]])
 
+    def test_delete(self):
+        """Correctly deletes a preprocessed data"""
+        # testing regular delete
+        ppd = PreprocessedData.create(
+            self.study, self.params_table,
+            self.params_id, self.filepaths, prep_template=self.prep_template,
+            ebi_submission_accession=self.ebi_submission_accession,
+            ebi_study_accession=self.ebi_study_accession)
+        PreprocessedData.delete(ppd.id)
```

Comment:
My feeling is that, since this is a test of functionality, it should
explicitly test the functionality. Since the code is supposed to delete
rows from multiple tables, the deletion of those rows should be explicitly
checked. This also means that if any future code does anything to deletion,
we can still make sure those rows are deleted.
On Apr 7, 2015 1:12 PM, "josenavas" <notifications@github.com> wrote:

> In qiita_db/test/test_data.py
> <https://github.com/biocore/qiita/pull/1042#discussion_r27915170>:
>
> > @@ -478,6 +478,38 @@ def test_create_data_type_only(self):
> >          # preprocessed_data_id, filepath_id
> >          self.assertEqual(obs, [[3, obs_id - 1], [3, obs_id]])
> >
> > +    def test_delete(self):
> > +        """Correctly deletes a preprocessed data"""
> > +        # testing regular delete
> > +        ppd = PreprocessedData.create(
> > +            self.study, self.params_table,
> > +            self.params_id, self.filepaths, prep_template=self.prep_template,
> > +            ebi_submission_accession=self.ebi_submission_accession,
> > +            ebi_study_accession=self.ebi_study_accession)
> > +        PreprocessedData.delete(ppd.id)
>
> @ElDeveloper <https://github.com/ElDeveloper> any thoughts on this? I'm
> fine with the code as it is. The reasons why I do not think that more
> checks are needed are:
>
>    - Everything is in a queue. If one statement fails, all of them fail.
>    - If one of the other tables are not removed, SQL will complain due to
>    FK restriction.
>
> I was just proposing an alternative solution, to see if @antgonza
> <https://github.com/antgonza> and @squirrelo
> <https://github.com/squirrelo> agree on something intermediate. @squirrelo
> <https://github.com/squirrelo> given these arguments, are you still
> feeling strong on put some extra checks in place?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/qiita/pull/1042/files#r27915170>.
>